index.js

```diff
@@ -240,13 +240,19 @@ FSWatcher.prototype._awaitWriteFinish = function(path, threshold, callback) {
   (function awaitWriteFinish (prevStat) {
     fs.stat(fullPath, function(err, curStat) {
       if (err) {
-        delete this._pendingWrites[path];
+        // prevent `unlink` race condition
+        this._getWatchedDir(sysPath.dirname(path))
+          .remove(sysPath.basename(path));
+        if (this._pendingWrites[path]) {  
+          this._pendingWrites[path].cancelWait();
```

Comment:
I did try that. Didn't solve it on its own. The last change in this PR seems to be working